### PR TITLE
use $GLOBALS['_composer_autoload_path'] if possible

### PR DESCRIPTION
**Pull request by Kanti**
I cloned infection into a local directory to install it via composer from there. 
So I can work on it in my normal project along side it. ( https://getcomposer.org/doc/05-repositories.md#path)

But than the autoloading is not found.
With the `$GLOBALS['_composer_autoload_path']` it is now possible to install it from anythere.
This is possible since composer 2.2:
https://getcomposer.org/doc/articles/vendor-binaries.md#finding-the-composer-autoloader-from-a-binary

### CODE REVIEW

File: bin/infection [hunk 1]
```diff
@@ -48,6 +48,13 @@ if (\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) === false) {
 
 // Infection autoloading
 (static function (): void {
+    if (\file_exists($autoload = ($GLOBALS['_composer_autoload_path'] ?? ''))) {
```
Comment (sanmai) [inline bin/infection:51]: What do you think about checking if the global exists before checking if the file exists?

So we don't call file_exists on an empty string.

File: bin/infection [hunk 2]
```diff
@@ -48,6 +48,14 @@ if (\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true) === false) {
 
 // Infection autoloading
 (static function (): void {
+    $autoload = $GLOBALS['_composer_autoload_path'] ?? '';
```
Comment (Kanti) [inline bin/infection:51]:
```suggestion
    $autoload = $GLOBALS['_composer_autoload_path'] ?? '';

```

**Comment by maks-rafalko** [comment]
> If you want to rely on this in your package you should however make sure to also require "composer-runtime-api": "^2.2" to ensure that the package gets installed with a Composer version supporting the feature.

I guess we should update package.json/lock to require ^2.2 then?

**Comment by Kanti** [comment]
We could, but we don't need to. If the user uses composer 2.2 or higher he can profit from this change. if he dose not, it still works, as we have the other 2 includes.

**Comment by maks-rafalko** [comment]
ok, there is a failing test unrelated to your changes. I will merge this PR as soon as I fix the tests.

**Review by maks-rafalko — APPROVED** [review]

**Comment by maks-rafalko** [comment]
Thank you @Kanti

**Pull request merged**